train/sft/toenet_base/train.py

```diff
@@ -128,8 +131,17 @@ def train_loop(
                 print("Validation Loss")
                 print(f"step {epoch_idx}&{step_idx}", val_loss)
 
+                early_stopping(val_loss)
+                if early_stopping.early_stop:
+                    print("Early stopping")
+                    break 
+
             global_step_counter += 1
 
+        if early_stopping.early_stop:
+            print("Early stopping")
+            break 
```

Comment:
why are you printing "early stopping" in both line 136 and 142?

Reply:
@superleesa 
Early stopping should occur if the validation loss decreases in **n consecutive epochs**? I'm wondering if I should write it in the nested loop or in the loop (one for loop)

Reply:
I think the problem is that we are calculating the validation too often. 

Can you change the code so that we only calculate the validation loss at the end of each epoch? Then we can determine to early stop after the validation loss computation.

Reply:
@superleesa 
okay, you mean we only need to calculate the validation loss one time every epoch right?

Reply:
yup